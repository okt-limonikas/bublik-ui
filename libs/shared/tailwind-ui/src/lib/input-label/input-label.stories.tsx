/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import { StoryFn, Meta } from '@storybook/react';
import { InputLabel, InputLabelProps } from './input-label';

export default {
	component: InputLabel,
	title: 'form/Label',
	decorators: [
		(Story) => <div className="p-4 bg-white rounded-md">{Story()}</div>
	]
} as Meta;

const Template: StoryFn<InputLabelProps> = (args) => (
	<InputLabel {...args}>Test Label</InputLabel>
);

export const Primary = {
	render: Template,
	args: {}
};
