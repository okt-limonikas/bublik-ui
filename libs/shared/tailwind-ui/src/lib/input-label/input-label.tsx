/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import { ComponentPropsWithRef, FC, forwardRef } from 'react';

import { cn } from '../utils';

export type InputLabelProps = ComponentPropsWithRef<'label'>;

export const InputLabel: FC<InputLabelProps> = forwardRef(
	({ children, className, ...props }, ref) => {
		return (
			<label
				className={cn(
					'font-normal text-text-secondary text-[0.875rem]',
					className
				)}
				{...props}
				ref={ref}
				data-testid="input-label"
			>
				{children}
			</label>
		);
	}
);
