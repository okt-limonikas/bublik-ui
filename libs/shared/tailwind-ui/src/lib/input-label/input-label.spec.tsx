/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import { it, describe, expect } from 'vitest';

import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { InputLabel } from './input-label';

describe('InputLabel', () => {
	it('should render successfully', () => {
		const { getByTestId } = render(<InputLabel />);

		const element = getByTestId('input-label');

		expect(element).toBeVisible();
	});

	it('should match snapshot', () => {
		const tree = renderer.create(<InputLabel />).toJSON();

		expect(tree).toMatchSnapshot();
	});
});
