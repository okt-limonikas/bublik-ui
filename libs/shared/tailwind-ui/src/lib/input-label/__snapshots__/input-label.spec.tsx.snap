// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputLabel > should match snapshot 1`] = `
<label
  className="font-normal text-text-secondary text-[0.875rem]"
  data-testid="input-label"
/>
`;
